### ISSUE TITLE
Basic auth configuration does not seem to work via config file

### ISSUE THREAD

**Issue by jgoeres** [helm/chartmuseum#308]
I would prefer using a config file instead of tons of command line args to configure Chartmuseum.
For our internal testing purposes, using basic auth with anonymous GET access should be fine, so I added these lines to my config file, 

    basic.auth.user: admin
    basic.auth.pass: s3cr37
    auth.anonymous.get: true

based on the documentation saying:

> When using a yaml file instead of the command line， Convert the - of the parameter in the command to .. Note that --storage is converted to storage.backend

Alas, it doesn't work, no authentication is active.

I also tried removing the "auth.anonymous.get" part, cause I was unsure what value to set here, since this is a command-line flag without an argument.

When I run chartmuseum and specify the basic auth options as command line args, it works as expected:

    ./chartmuseum --config chartmuseum.yaml --basic-auth-user admin --basic-auth-pass admin --auth-anonymous-get

Other settings from my config file (port, storage.backend etc.) are working fine.

I also tried leaving out 
But

**Comment by jdolitsky** [comment]
@jgoeres it seems that the strings have an extra "." character. I realize this is misleading since the flags have "-" characters.

Please see all the keys here: https://github.com/helm/chartmuseum/blob/969515a51413e1f1840fb99509401aa3c63deccd/pkg/config/vars.go#L169

You should be able to do
```
basicauth.user: admin
basicauth.pass: s3cr37
```

Or alternatively:
```
basicauth: 
  user: admin
  pass: s3cr37
```

Please let me know if that works out

**Comment by jgoeres** [comment]
Hi,
thanks, the basicauth.user and basicauth.pass keys work inthe config file, and in vars.go I found the equivalent of the switch "--auth-anonymous-get" to be 

    authanonymousget: true

Maybe the "mapping rule" mentioned in the docs that I quoted above should be removed entirely and replaced with a pointer to the vars.go file, on perspective to be replaced with a table of all command line flags and their corresponding config file key?

Regards

J

**Comment by jayme-github** [comment]
I would like as well to configure chartmuseum via config file and while doing so, found that a lot of inconsistencies the configVars keys used. For a lot of commands the `-` has to be omitted and there also are special cases like: https://github.com/helm/chartmuseum/blob/a258b600c1dea4fe536e96ead74716b13e1bff66/pkg/config/vars.go#L733 and https://github.com/helm/chartmuseum/blob/a258b600c1dea4fe536e96ead74716b13e1bff66/pkg/config/vars.go#L109 

I ended up using something like the following to generate a valid config file with everything set to defaults:
```go
package main

import (
        "fmt"

        "github.com/spf13/afero"
        "helm.sh/chartmuseum/pkg/config"
)

func main() {
        c := config.NewConfig()
        memFS := afero.NewMemMapFs()
        c.Viper.SetFs(memFS)
        c.WriteConfigAs("/tmp/foo.yaml")
        b, _ := afero.ReadFile(memFS, "/tmp/foo.yaml")
        fmt.Print(string(b))
}
```

**Comment by jdolitsky** [comment]
@jayme-github lmk if this is OK to close

**Comment by jayme-github** [comment]
> @jayme-github lmk if this is OK to close

I think this particular issue is okay to close. But the general problem still exists of cause.
There are a couple of parameters where the `Convert the - of the parameter in the command to .` rule from README.md does not apply. See my ragged attempt to fix this in #343 
Maybe we should have a more general issue for this.

**Comment by jdolitsky** [comment]
@jayme-github ok, ty. Going to close, but feel free to express the more general problem in a new issue if necessary
